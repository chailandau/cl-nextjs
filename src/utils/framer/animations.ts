export const menuAnimations = (prefersReducedMotion?: boolean) => ({
    closed: {
        y: '20%',
        opacity: 0,
        transition: {
            delay: prefersReducedMotion ? 0 : 0.15,
            length: prefersReducedMotion ? 0 : 1
        }
    },
    open: {
        y: 0,
        opacity: 1,

        transition: {
            type: 'spring',
            duration: prefersReducedMotion ? 0 : 0.4
        }
    }
});

export const iconFlip = {
    closed: { rotate: 0, transition: { type: 'spring', duration: 0.25 } },
    open: {
        rotate: 180,
        transition: {
            type: 'spring',
            duration: 0.25
        }
    }
};
export const backToTop = (prefersReducedMotion?: boolean) => ({
    visible: {
        y: 0,
        opacity: 1,
        transition: { duration: prefersReducedMotion ? 0 : 0.6 }
    },
    hidden: {
        y: 100,
        opacity: 0,
        transition: { duration: prefersReducedMotion ? 0 : 0.6 }
    },
    hovered: {
        scale: prefersReducedMotion ? 1 : 1.2,
        transition: { duration: prefersReducedMotion ? 0 : 0.2 }
    },
    tapped: {
        scale: 1
    }
});
