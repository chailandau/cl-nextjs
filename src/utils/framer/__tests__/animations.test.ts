import * as framerAnimations from '../animations';

describe('Framer animations', () => {
    Object.keys(framerAnimations).forEach((animationName) => {
        const animation =
            framerAnimations[animationName as keyof typeof framerAnimations];
        it(`${animationName} matches snapshot`, () => {
            expect(animation).toMatchSnapshot();
        });
    });
});
