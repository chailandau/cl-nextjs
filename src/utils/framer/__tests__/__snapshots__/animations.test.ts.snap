// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Framer animations iconFlip matches snapshot 1`] = `
{
  "closed": {
    "rotate": 0,
    "transition": {
      "duration": 0.25,
      "type": "spring",
    },
  },
  "open": {
    "rotate": 180,
    "transition": {
      "duration": 0.25,
      "type": "spring",
    },
  },
}
`;

exports[`Framer animations menuAnimations matches snapshot 1`] = `
{
  "closed": {
    "opacity": 0,
    "transition": {
      "delay": 0.15,
      "length": 1,
    },
    "y": "20%",
  },
  "open": {
    "opacity": 1,
    "transition": {
      "duration": 0.4,
      "type": "spring",
    },
    "y": 0,
  },
}
`;
